src/routes/article-list.ts

```diff
@@ -1,34 +1,47 @@
+import clownface, { Clownface } from 'clownface';
+import { constants } from 'http2';
 import all from 'it-all';
-import { JsonLdObj } from 'jsonld/jsonld-spec';
 import { Next } from 'koa';
-import { BlankNode } from 'rdf-js';
-import { hydra, rdf, schema } from 'rdf-namespaces';
+import { addAll } from 'rdf-dataset-ext';
+import { BlankNode, DatasetCore } from 'rdf-js';
+import { toRdf } from 'rdf-literal';
+import url from 'url';
 import { AppContext, AppMiddleware } from '../app';
+import { hydra, rdf, schema } from '../namespaces';
 import Routes from './index';
 
 export default (): AppMiddleware => (
   async ({
-    articles, request, response, router,
+    dataFactory: { literal, namedNode }, articles, request, response, router,
   }: AppContext, next: Next): Promise<void> => {
-    const [list, count] = await Promise.all([all(articles), articles.count()]);
+    const graph = clownface({
+      dataset: response.dataset,
+      term: namedNode(url.resolve(request.origin, router.url(Routes.ArticleList))),
+    });
 
-    response.body = {
-      '@context': {
-        '@base': request.origin,
-      },
-      '@id': router.url(Routes.ArticleList),
-      '@type': hydra.Collection,
-      [hydra.title]: { '@value': 'List of articles', '@language': 'en' },
-      'http://www.w3.org/ns/hydra/core#manages': {
-        'http://www.w3.org/ns/hydra/core#property': { '@id': rdf.type },
-        'http://www.w3.org/ns/hydra/core#object': { '@id': schema.Article },
-      },
-      [hydra.totalItems]: count,
-      [hydra.member]: {
-        '@list': list.map((parts: [BlankNode, JsonLdObj]): JsonLdObj => parts[1]),
-      },
-    };
-    response.type = 'jsonld';
+    const listPromise = all(articles)
+      .then((list): void => {
+        list.forEach(([id, article]: [BlankNode, DatasetCore]): void => {
+          graph.addOut(hydra.member, id);
+          addAll(graph.dataset, article);
+        });
+      });
+
+    const countPromise = articles.count()
+      .then((count): void => {
```

Comment:
use of multiple Promises for parallelism rather than `await`ing for their result?

Reply:
Yeah, micro-optimisation but means processing based on those values can happen immediately rather than waiting for both calls to finish.